### ISSUE TITLE
Rating: keyboard navigation is broken

### ISSUE THREAD

**Issue by JaMeZ-B** [primefaces/primeng#17315]
### Describe the bug

As you can witness in the docs at https://primeng.org/rating the keyboard navigation does not work as intended. Funnily, there seems to be mixup, as **keyboard navigation is possible in `readonly` mode**!

### Reason for not contributing a PR

- [X] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/github-3l7sgfgn?file=src%2Fapp%2Fapp.component.html

### Environment

macOS 15.2

### Angular version

19

### PrimeNG version

v19

### Steps to reproduce the behavior

Press 'Tab' until the component is focused, then try to change the value using the arrow keys.

### Expected behavior

If readonly is `false` the keyboard navigation should work as specified in the docs. If it is set to true, it should _not_ work.
